src/NewTwitchApi/Resources/EntitlementsApi.php

```diff
@@ -20,4 +21,38 @@ public function getEntitlementGrantsUploadURL(string $bearer, string $manifestId
 
         return $this->postApi('entitlements/upload', $bearer, $queryParamsMap);
     }
+
+    /**
+     * @throws GuzzleException
+     * @link https://dev.twitch.tv/docs/api/reference#get-code-status
+     */
+    public function getStatusCode(string $bearer, array $codes = [], int $userId): ResponseInterface
+    {
+        $queryParamsMap = [];
+
+        foreach ($codes as $code) {
+            $queryParamsMap[] = ['key' => 'code', 'value' => $code];
+        }
+
+        $queryParamsMap[] = ['key' => 'user_id', 'value' => $userId];
+
+        return $this->callApi('entitlements/codes', $bearer, $queryParamsMap);
+    }
+
+    /**
+     * @throws GuzzleException
+     * @link https://dev.twitch.tv/docs/api/reference#redeem-code
+     */
+    public function redeemCode(string $bearer, array $codes = [], int $userId): ResponseInterface
```

Comment:
Required arguments should come before optional arguments.

Reply:
@echosa neither of these are optional, this will return an exception of no codes are passed, but it allows up to 20 entries. I think this is fine - please confirm if you still want it changed?

Reply:
Regardless of the Twitch API, the way this method is defined, `$codes` is optional, because a default of `[]` is supplied, whereas `$userId` is required, without a default value. Please rearrange them so that required parameters are first.